### ISSUE TITLE
Camvid

### ISSUE THREAD

**Issue by miscedence12** [RolandGao/RegSeg#12]
@RolandGao Hello,I  wonder to know whether you will publish the weights file trained on the Camvid dataset.

**Comment by RolandGao** [comment]
I don't have the CamVid weights anymore. The Cityscapes weights are much more useful because Cityscapes is a larger dataset. You can train on the CamVid dataset yourself by preparing the dataset, downloading the Cityscapes trainval pretrained weights, and using the training config here (https://github.com/RolandGao/RegSeg/blob/main/configs/camvid_200epochs.yaml).

**Comment by miscedence12** [comment]
@RolandGao  hello，Can you provide me with the directory structure of the Camvid dataset you use and show sample images? The data set I prepared by myself seems to be unable to complete the training,thanks!

**Comment by RolandGao** [comment]
CamVid does require a bit of preprocessing before it can be used. The labels are converted from rgb to class labels first using something like the rgb_mask2 function in camvid.py. The directory structure is 
-- train
-- train_labels
-- val
-- val_labels
-- test
-- test_labels
-- class_dict.csv

**Comment by miscedence12** [comment]
@RolandGao  Thanks for your reply,I have made the following changes to camvid.py. Do you think it is right?  My label is rgb image.
 line43:     img = Image.open(self.images[index]).convert('RGB')
line44:        target = rgb_to_mask2(Image.open(self.masks[index]),color_to_class=self.color_to_class)

**Comment by RolandGao** [comment]
I think it should be correct, but you have to test it out yourself to confirm.

**Comment by miscedence12** [comment]
@RolandGao  I've already verified it. It's right；Recently, I want to train the model on  coco dataset. Have you ever trained it before? If so, is there any pretrained weights? 

Best wishes!
Jie

**Comment by RolandGao** [comment]
Hi,

Training on coco should be supported by the repository with the "coco_100epochs.yaml" config file, but I don't have pretrained weights.

**Comment by anizyu** [comment]
Do I need to rewrite a training file to train the Camvid dataset?

**Comment by RolandGao** [comment]
No, you can use the same train.py. You only need to prepare the dataset.

**Comment by ZDrawer** [comment]
> @RolandGao I've already verified it. It's right；Recently, I want to train the model on coco dataset. Have you ever trained it before? If so, is there any pretrained weights?我已经验证过了。没错;最近，我想在 coco 数据集上训练模型。你以前训练过它吗？如果有，是否有任何预训练的权重？
> 
> Best wishes! 愿你安好！ Jie 他们

Hello, I use the camvid dataset for training, but I keep getting an error, I would like to ask if you can provide the processed camvid dataset

**Comment by miscedence12** [comment]
> > @RolandGao I've already verified it. It's right；Recently, I want to train the model on coco dataset. Have you ever trained it before? If so, is there any pretrained weights?我已经验证过了。没错;最近，我想在 coco 数据集上训练模型。你以前训练过它吗？如果有，是否有任何预训练的权重？
> > Best wishes! 愿你安好！ Jie 他们
> 
> Hello, I use the camvid dataset for training, but I keep getting an error, I would like to ask if you can provide the processed camvid dataset

ok，you can give me your email，i will send it to you later.

**Comment by ZDrawer** [comment]
这是来自QQ邮箱的假期自动回复邮件。您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by ZDrawer** [comment]
> > > @RolandGao I've already verified it. It's right；Recently, I want to train the model on coco dataset. Have you ever trained it before? If so, is there any pretrained weights?我已经验证过了。没错;最近，我想在 coco 数据集上训练模型。你以前训练过它吗？如果有，是否有任何预训练的权重？
> > > Best wishes! 愿你安好！ Jie 他们
> > 
> > 
> > Hello, I use the camvid dataset for training, but I keep getting an error, I would like to ask if you can provide the processed camvid dataset
> 
> ok，you can give me your email，i will send it to you later.

My email address is: 2011239118@qq.com. Thank you very much for your reply and help.
